sklearn/model_selection/tests/test_split.py

```diff
@@ -0,0 +1,717 @@
+    # folds are not possible
+    X2 = np.ones((17, 2))
+    kf = KFold(3)
+    check_cv_coverage(kf, X2, y=None, labels=None, expected_n_iter=3,
+                      n_samples=17)
+
+
+def test_kfold_no_shuffle():
+    # Manually check that KFold preserves the data ordering on toy datasets
+    X2 = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]]
+
+    splits = KFold(2).split(X2[:-1])
+    train, test = next(splits)
+    assert_array_equal(test, [0, 1])
+    assert_array_equal(train, [2, 3])
+
+    train, test = next(splits)
+    assert_array_equal(test, [2, 3])
+    assert_array_equal(train, [0, 1])
+
+    splits = KFold(2).split(X2)
+    train, test = next(splits)
+    assert_array_equal(test, [0, 1, 2])
+    assert_array_equal(train, [3, 4])
+
+    train, test = next(splits)
+    assert_array_equal(test, [3, 4])
+    assert_array_equal(train, [0, 1, 2])
+
+
+def test_stratified_kfold_no_shuffle():
+    # Manually check that StratifiedKFold preserves the data ordering as much
+    # as possible on toy datasets in order to avoid hiding sample dependencies
+    # when possible
+    X, y = np.ones((4, 2)), [1, 1, 0, 0]
+    splits = StratifiedKFold(2).split(X, y)
+    train, test = next(splits)
+    assert_array_equal(test, [0, 2])
+    assert_array_equal(train, [1, 3])
+
+    train, test = next(splits)
+    assert_array_equal(test, [1, 3])
+    assert_array_equal(train, [0, 2])
+
+    splits = StratifiedKFold(2).split(np.ones((7, 2)), [1, 1, 1, 0, 0, 0, 0])
+    train, test = next(splits)
+    assert_array_equal(test, [0, 1, 3, 4])
+    assert_array_equal(train, [2, 5, 6])
+
+    train, test = next(splits)
+    assert_array_equal(test, [2, 5, 6])
+    assert_array_equal(train, [0, 1, 3, 4])
+
+
+def test_stratified_kfold_ratios():
+    # Check that stratified kfold preserves label ratios in individual splits
+    # Repeat with shuffling turned off and on
+    n_samples = 1000
+    X = np.ones(n_samples)
+    labels = np.array([4] * int(0.10 * n_samples) +
+                      [0] * int(0.89 * n_samples) +
+                      [1] * int(0.01 * n_samples))
+
+    for cv in [StratifiedKFold(5, shuffle=s) for s in (True, False)]:
+        for train, test in cv.split(X, y=labels):
+            assert_almost_equal(np.sum(labels[train] == 4) / len(train), 0.10,
+                                2)
+            assert_almost_equal(np.sum(labels[train] == 0) / len(train), 0.89,
+                                2)
+            assert_almost_equal(np.sum(labels[train] == 1) / len(train), 0.01,
+                                2)
+            assert_almost_equal(np.sum(labels[test] == 4) / len(test), 0.10, 2)
+            assert_almost_equal(np.sum(labels[test] == 0) / len(test), 0.89, 2)
+            assert_almost_equal(np.sum(labels[test] == 1) / len(test), 0.01, 2)
+
+
+def test_kfold_balance():
+    # Check that KFold returns folds with balanced sizes
+    for i in range(11, 17):
+        kf = KFold(5).split(X=np.ones(i))
```

Comment:
I see here you use a 1d X for split, yet in other previous tests you used 2d Xs. Could you add a test that ensures that 1d Xs work as expected, and then use 1d Xs in all tests? It'd reduce the signal to noise ratio.